### PR TITLE
SUP-1400: Test suite and test suite team (team suite) resource retries 

### PR DESCRIPTION
**Pull request by james2791**
feat (Timeouts/Retries): Add timeouts (provider config set) and CRUD retries to the Test suite and test suite team resources.

## PR checklist:
- [x] `docs/` updated (provider docs updated previously)
- [x] tests added (TST tests migrated to t.Run())
- [x] an example added to `examples/` (provider example updated with `timeouts` block)
- [x] `CHANGELOG.md` updated with pending release information

Also converted the Test suite team tests with `t.Run` to enable parallelism / set timeouts in test configuration. Both TS and TST have randomness enabled previously.

This PR also adjusts the Client's `makeRequest` function to returns errors with a `returned error` format line so that REST API requests can benefit from the rate limit/server error retry methods implemented.

### CODE REVIEW

File: buildkite/resource_test_suite.go [hunk 1]
```diff
@@ -54,19 +55,45 @@ func (ts *testSuiteResource) Create(ctx context.Context, req resource.CreateRequ
...
+
 	if resp.Diagnostics.HasError() {
 		return
 	}
 
+	timeout, diags := ts.client.timeouts.Create(ctx, DefaultTimeout)
+	resp.Diagnostics.Append(diags...)
+
 	// The REST API requires team UUIDs but everything else in the provider uses GraphQL IDs. So we map from UUID to ID
 	// here
```
Comment (mcncl) [inline buildkite/resource_test_suite.go:69]: Perhaps this should be moved to where it is relevant?

File: buildkite/resource_test_suite.go [hunk 2]
```diff
@@ -78,10 +105,26 @@ func (ts *testSuiteResource) Create(ctx context.Context, req resource.CreateRequ
...
+
+		if err != nil {
+			if isRetryableError(err) {
+				return retry.RetryableError(err)
+			}
+			return retry.NonRetryableError(err)
+		}
+
+		return nil
+	})
```
Comment (jradtilbrook) [inline buildkite/resource_test_suite.go:121]: I wonder if the whole thing should be a single retry? Maybe not, I don't really know but calling it out if anyone has an argument either way

File: buildkite/resource_test_suite.go [hunk 3]
```diff
@@ -119,20 +183,48 @@ func (*testSuiteResource) Metadata(ctx context.Context, req resource.MetadataReq
...
+
 	if err != nil {
-		resp.Diagnostics.AddError("Failed to load test suite from GraphQL", err.Error())
+		resp.Diagnostics.AddError(
+			"Failed to load test suite from GraphQL",
+			err.Error(),
+		)
 		return
```
Comment (jradtilbrook) [inline buildkite/resource_test_suite.go:222]: I don't think this has the remove from state logic like we've been adding everywhere. Can you add it?

**Review by jradtilbrook — CHANGES_REQUESTED** [review]
Looks good. Just some minor things

File: buildkite/resource_test_suite.go (hunk 2 above)
Comment (james2791) [reply buildkite/resource_test_suite.go:121]: Fair point - I _think_ it makes sense leaving them seperate, as the first technically is a read, the second is the actual creation. Latest commit has the first using the `Read` timeout, and the second (REST POST) using the `Create` timeout via the provider

Comment (james2791) [reply buildkite/resource_test_suite.go:121]: Fair point - I think it makes sense leaving them seperate, as the first technically is a read, the second is the actual creation. Latest commit has the first using the Read timeout, and the second (REST POST) using the Create timeout via the provider

File: buildkite/resource_test_suite.go [hunk 4]
```diff
@@ -54,19 +55,45 @@ func (ts *testSuiteResource) Create(ctx context.Context, req resource.CreateRequ
...
 
-	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	diags := req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+
 	if resp.Diagnostics.HasError() {
 		return
 	}
 
+	timeout, diags := ts.client.timeouts.Read(ctx, DefaultTimeout)
```
Comment (jradtilbrook) [inline buildkite/resource_test_suite.go:65]: Hmm I think this should still use the create timeout value. Or at least the read part of this create function use the read timeout and the create part of it use the create part

**Review by jradtilbrook — CHANGES_REQUESTED** [review]

File: buildkite/resource_test_suite.go (hunk 4 above)
Comment (james2791) [reply buildkite/resource_test_suite.go:66]: Just whipped that in wit the latest commit - `POST` to test suite creation endpoint using the `Create` timeout - above with Team owner UUID -> ID using the `Read` T/O

**Review by jradtilbrook — APPROVED** [review]

**Review by jradtilbrook — DISMISSED** [review]

**Review by jradtilbrook — APPROVED** [review]

**Pull request merged**